src/components/PopupWithForm.js

```diff
@@ -20,12 +21,25 @@ class PopupWithForm extends Popup {
     });
     return inputValue;
   }
+  setInputValues(inputValue) {
+    this._inputList.forEach((inputEl) => {
+      inputEl.value = inputValue[inputEl.name];
+    });
+  }
+
+  renderLoading(isLoading, loadingText = "Saving..") {
+    if (isLoading) {
+      this._formButton.textContent = loadingText;
+    } else {
+      this._formButton.textContent = this._formButtonText;
+    }
+  }
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (e) => {
       e.preventDefault();
       this._handleFormSubmit(this._getInputValues());
-      this.close();
+      this._popupForm.reset();
```

Comment:
`this._popupForm.reset();` 


You need to **reset** the form only in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.